server/service/integration_mdm_test.go

```diff
@@ -10418,6 +10475,15 @@ func (s *integrationMDMTestSuite) TestVPPApps() {
 		})
 	}
 
+	/// Attempt self-service install
+	// Not available for self-service
+	var ssInstallResp submitSelfServiceSoftwareInstallResponse
+	s.DoJSON("POST", fmt.Sprintf("/api/latest/fleet/device/%s/software/install/%d", selfServiceToken, iPadOSTitleID), &fleetSelfServiceSoftwareInstallRequest{},
+		http.StatusBadRequest, &ssInstallResp)
+	// Successful install
+	s.DoJSON("POST", fmt.Sprintf("/api/latest/fleet/device/%s/software/install/%d", selfServiceToken, macOSTitleID), &fleetSelfServiceSoftwareInstallRequest{},
+		http.StatusAccepted, &ssInstallResp)
```

Comment:
Looks like this would be a good place to verify the activity, just so that self-service activity payload is tested (unless it's already elsewhere, I may have missed it).

Reply:
There's actually no activity for associating a VPP app with a machine. I looked everywhere for one but couldn't find anything. The only activity that currently exists is for assigning a VPP app to a team. I can bring this up later but it will also require a front-end component

Reply:
I was thinking of this: https://github.com/fleetdm/fleet/blob/08763d59f1dd28c95ad133a7fbb666e27810ed3c/server/service/apple_mdm.go#L2764-L2782

I.e. the `ActivityInstalledAppStoreApp` but I just now noticed that it hadn't been updated to include the `self_service` flag (to indicate it was self-installed or not). Presumably we want that activity to match the software installers one?